runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4185,6 +4185,37 @@ class INTERPRETER_CLASS
 
 		return rc;
 	}
+
+	VMINLINE VM_BytecodeAction
+	inlUnsafeIsFieldAtOffsetFlattened(REGISTER_ARGS_LIST)
+	{
+		I_64 offset = *(I_64*)_sp;
+		j9object_t clz = *(j9object_t*)(_sp + 2);
+
+		I_32 isFlattened = (I_32)FALSE;
+
+		if (NULL != clz) {
+			J9Class *clzJ9Class = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, clz);
+			if ((!J9CLASS_IS_ARRAY(clzJ9Class)) && (NULL != clzJ9Class->flattenedClassCache)) {
+
+				/* caller provided offset will include header size but entry->offset does not */
+				offset -= (I_64)J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);
+
+				UDATA numberOfFlattenedFields = clzJ9Class->flattenedClassCache->numberOfEntries;
+				for (UDATA i = 0; i < numberOfFlattenedFields; i++) {
+					J9FlattenedClassCacheEntry *entry = J9_VM_FCC_ENTRY_FROM_CLASS(clzJ9Class, i);
+					if (entry->offset == (UDATA)offset) {
```

Comment:
Added a conversion to `UDATA` here (it was only a compilation error on linux for some reason)

Reply:
You are doing `offset -= (I_64)J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);` before, 
probably better to convert `entry->offset` to (I_64) in the comparison.

Reply:
okay, changed in d9d1d40